File: 015/015.js
function binomialCoefficient(n, k) {
  if (k < 0 || k > n)
    return 0
  if (k === 0 || k === n)
    return 1
  x =  n - k < k ? n - k : k
  let c = 1
  for (let i = 0; i < x; i++)
    c = c * (n - i) / (i + 1)
  return c
}

console.log(binomialCoefficient(40, 20))

File: 015/015.rs
fn binomial_coefficient(n: u64, k: u64) -> u64 {
    if k > n {
        return 0;
    }
    if k == n {
        return 1;
    }
    let x = if n - k < k { n - k } else { k };
    let mut c: u64 = 1;
    for i in 0..x {
        c = c * (n - i) / (i + 1);
    }
    return c;
}

fn main() {
    print!("{}\n", binomial_coefficient(40, 20));
}
